lib/omnibus/software.rb

```diff
@@ -1077,15 +1080,26 @@ def path_key
     # Apply overrides in the @overrides hash that mask instance variables
     # that are set by parsing the DSL
     #
-    def apply_overrides(attr)
+    def apply_overrides(attr, override=overrides[attr])
       val = instance_variable_get(:"@#{attr}")
-      if val.is_a?(Hash) || overrides[attr].is_a?(Hash)
+      if val.is_a?(Hash) || override.is_a?(Hash)
         val ||= {}
-        override = overrides[attr] || {}
+        override ||= {}
         val.merge(override)
       else
-        overrides[attr] || val
+        override || val
+      end
+    end
+
+    #
+    # Transform github -> git in source
```

Comment:
Should make a note that this function modifies the given hash. I would suggest renaming to `canonicalize_source!` or duping the source hash before deleting a value.